File: backend/app/Models/UserClass.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class UserClass extends Model
{
    use HasFactory;

    protected $table = 'user_classes';
    protected $primaryKey = 'id_userclass';
    public $incrementing = true;
    protected $keyType = 'int';

    protected $fillable = [
        'fk_id_user',
        'fk_id_class',
        'role',
        'joined_at',
    ];
}

File: sea_app/lib/models/user_class.dart
class UserClass {
  final String idUserClass;
  final DateTime joinedAt;
  final String fkIdClass;
  final String fkIdUser;
  final DateTime createAt;
  final DateTime updateAt;

  UserClass({
    required this.idUserClass,
    required this.joinedAt,
    required this.fkIdClass,
    required this.fkIdUser,
    required this.createAt,
    required this.updateAt,
  });
}